### PR TITLE
Improve checker

### PR DESCRIPTION
**Pull request by jtydhr88**

### CODE REVIEW

**Comment by gamerson** [comment]
hey @jtydhr88  looks like three test failures:
```
All Failed Tests

Test Name
Duration
Age
 com.liferay.blade.test.apichanges.PortletConfigGetFormatMethodsTest.unicodeLanguageImplFile	0.17 sec	1
 com.liferay.blade.test.apichanges.PortletConfigGetFormatMethodsTest.test	0.21 sec	1
 com.liferay.blade.test.apichanges.PortletConfigGetFormatMethodsTest.test2
```

Each one of these says that more problems were found that were expected.  So this is likely due to the improved checker but can you update the tests as well.

File: com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java [hunk 1]
```diff
@@ -592,4 +594,57 @@ protected File getFile() {
 		return null;
 	}
 
+	private boolean typeMatch(String expectType, String paramType) {
+		boolean match = false;
+
+		if (expectType.endsWith(paramType) || paramType.endsWith(expectType)) {
+			match = true;
+		}
+		else if (expectType.equals("Object") || expectType.equals("java.lang.Object") || expectType.equals("T")
```
Comment (gamerson) [inline com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java]: What are expectedType == "T" and expectType == E and expectType == U?

File: com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java [hunk 2]
```diff
@@ -592,4 +594,57 @@ protected File getFile() {
...
+				match = true;
+		}
+		else if (expectType.equals("Long") || expectType.equals("java.lang.Long")) {
+			if (paramType.equals("long") || paramType.equals("Long") || paramType.equals("java.lang.Long")) {
+				match = true;
+			}
+		}
+		else if (expectType.equals("int")) {
+			if (paramType.equals("int") || paramType.equals("Integer") || paramType.equals("java.lang.Integer")
+					|| paramType.equals("short") || paramType.equals("btye"))
```
Comment (gamerson) [inline com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java]: "short" and "byte" aren't the same thing as "int" so does this really work?

Comment (gamerson) [inline com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java]: 'btye' is missspelled it should be 'byte'

File: com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java [hunk 3]
```diff
@@ -592,4 +594,57 @@ protected File getFile() {
...
+			if (paramType.equals("short") || paramType.equals("Short") || paramType.equals("java.lang.Short")
+					|| paramType.equals("btye"))
+				match = true;
+		}
+		else if (expectType.equals("Short") || expectType.equals("java.lang.Short")) {
+			if (paramType.equals("short") || paramType.equals("Short") || paramType.equals("java.lang.Short")) {
+				match = true;
+			}
+		}
+		else if (expectType.equals("btye")) {
```
Comment (gamerson) [inline com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java]: byte is misspelled

File: com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java [hunk 4]
```diff
@@ -592,4 +594,57 @@ protected File getFile() {
...
+			if (paramType.equals("btye") || paramType.equals("Btye") || paramType.equals("java.lang.Btye"))
+				match = true;
+		}
+		else if (expectType.equals("Btye") || expectType.equals("java.lang.Btye")) {
+			if (paramType.equals("btype") || paramType.equals("Btye") || paramType.equals("java.lang.Btye")) {
```
Comment (gamerson) [inline com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java]: spelling

Comment (gamerson) [inline com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java]: Here you check for paramType = 'java.lang.Byte' but on line 619 you don't.  If one is needed here, the other should be too right?  

Also we need more test coverage, we need some specific java classes showing the problems and showing where these new code blocks match it.

**Comment by gamerson** [comment]
hey @jtydhr88  can you make the suggested changes?  thanks!

**Comment by jtydhr88** [comment]
Yes, I am still working on it and I will resent next week

**Pull request closed without merging**